state-chain/pallets/cf-elections/src/electoral_systems/egress_success.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+	electoral_system::{
+		AuthorityVoteOf, ElectionIdentifierOf, ElectionReadAccess, ElectionWriteAccess,
+		ElectoralSystem, ElectoralWriteAccess, VotePropertiesOf,
+	},
+	vote_storage::{self, VoteStorage},
+	CorruptStorageError,
+};
+use cf_primitives::AuthorityCount;
+use cf_utilities::{all_same, success_threshold_from_share_count};
+use frame_support::{
+	pallet_prelude::{MaybeSerializeDeserialize, Member},
+	Parameter,
+};
+use sp_std::vec::Vec;
+
+/// This electoral system detects if something occurred or not. Voters simply vote if something
+/// happened, and if they haven't seen it happen, they don't vote.
+pub struct EgressSuccess<Identifier, Value, Settings, Hook> {
+	_phantom: core::marker::PhantomData<(Identifier, Value, Settings, Hook)>,
+}
+
+pub trait OnEgressSuccess<Identifier, Value> {
+	fn on_egress_success(id: Identifier, value: Value);
+}
```

Comment:
This is basically identical to the `OnChangeHook` from the `change` elecoral system. 

Do we need another trait?

Reply:
I think it just makes it clear that this isn't called on a change, but only once.

Reply:
This electoral system is much more general than just egresses... but I couldn't think of a good general name for it. It's effectively "WatchForParticularThing"

Reply:
Yeah, not saying we need to change this now, just noting that there seems to be a lot of repetition / boilerplate.